From 3dfdc23fb297c506913ef555f6b0f5c8ba1504ab Mon Sep 17 00:00:00 2001
Message-Id: <3dfdc23fb297c506913ef555f6b0f5c8ba1504ab.1431284209.git.chang-joon.lee@intel.com>
In-Reply-To: <89460f3fc3d6c0659d44897a6adec3c3fecc5564.1431284209.git.chang-joon.lee@intel.com>
References: <89460f3fc3d6c0659d44897a6adec3c3fecc5564.1431284209.git.chang-joon.lee@intel.com>
From: John Harrison <John.C.Harrison@Intel.com>
Date: Fri, 27 Mar 2015 18:27:13 +0000
Subject: [PATCH 46/68] SQUASHME! [VPG]: drm/i915: Add fences to scheduler
 dump

When things go wrong with native syncs, it can be useful to see what
the scheduler/sync connection is. This patch adds the native sync
status of each node to the scheduler's status dump.

This patch could be squashed in to the one that originally added the
scheduler status dump facility:
  commit 557478d5fbdded9a79fd09c6f021ffbdb9dfe9ce
  Author: John Harrison <John.C.Harrison@Intel.com>
  Date:   Thu Apr 24 15:40:39 2014 +0100
  Subject: FOR_UPSTREAM [VPG]: drm/i915: Added debug state dump facilities to scheduler
  Change-Id: I0634168e3f3465ff023f5a673165c90b07e535b6

Change-Id: I5c9928c47b9cb68bce7788e60f2ef7fb6bf27883
For: VIZ-1587
For: VIZ-4741
Signed-off-by: John Harrison <John.C.Harrison@Intel.com>
---
 drivers/gpu/drm/i915/i915_scheduler.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_scheduler.c b/drivers/gpu/drm/i915/i915_scheduler.c
index 0357915..9801c59 100644
--- a/drivers/gpu/drm/i915/i915_scheduler.c
+++ b/drivers/gpu/drm/i915/i915_scheduler.c
@@ -948,7 +948,7 @@ int i915_scheduler_dump_locked(struct intel_engine_cs *ring, const char *msg)
 					deps++;
 
 			DRM_DEBUG_DRIVER("<%s>   %c:%02d> index = %d, uniq = %d, seqno"
-					 " = %d/%s, deps = %d / %d, %s [pri = "
+					 " = %d/%s, deps = %d / %d, fence = %p/%d, %s [pri = "
 					 "%4d]\n", ring->name,
 					 i915_scheduler_queue_status_chr(node->status),
 					 count,
@@ -957,6 +957,8 @@ int i915_scheduler_dump_locked(struct intel_engine_cs *ring, const char *msg)
 					 node->params.request->seqno,
 					 node->params.ring->name,
 					 deps, node->num_deps,
+					 node->params.fence_wait,
+					 node->params.fence_wait ? node->params.fence_wait->status : 0,
 					 i915_qe_state_str(node),
 					 node->priority);
 
-- 
1.7.9.5

